org.eclipse.lsp4e/plugin.xml

```diff
@@ -282,6 +282,27 @@
             </visibleWhen>
          </command>
       </menuContribution>
+      <menuContribution
+            allPopups="false"
+            locationURI="menu:org.eclipse.ui.views.ContentOutline">
+         <menu
+               id="org.eclipse.lsp4e.outline.hideMenu"
+               label="Hide...">
```

Comment:
If we can make it consistent with the JDT, I think that would be good. That would mean having a label such as

```suggestion
               label="Filter Elements...">
```

and the in the page a header such as "Select the elements to exclude from the view:". Do you think that would be a small improvement?

Reply:
I'm not sure with this point. First, I thought, we should the same kind of filtering as in JDT, but due to limitations in the LSP specification that will not work (e.g. we cannot hide static or non-public members since `DocumentSymbol` from LSP4J and LSP spec provides no such details). Since our filters are a lot simpler than those from JDT, I thought, it would be a good idea ti simplify the dialog compared to JDT.

Besides, it would be more effort to implement the same dialog as in JDT, but IMHO it would not offer any improvement to the user.

If someday LSP gets extended to provide more details about access modifiers and similar details (see https://github.com/microsoft/language-server-protocol/issues/98), then we could change the dialog to look the same as in JDT and also add toolbar icons, e.g. for hiding non-public members, etc.

Do you agree?

Reply:
Yes, I agree